daemon/util.h

```diff
@@ -0,0 +1,181 @@
+ *      distribution.
+ *   3. Neither the names of the copyright holders nor the names of its
+ *      contributors may be used to endorse or promote products derived
+ *      from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * HOLDERS OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * The views and conclusions contained in the software and documentation
+ * are those of the authors and should not be interpreted as representing
+ * any official policies, either expressed or implied.
+ */
+
+#ifndef  UTIL_H_
+# define UTIL_H_
+
+# include <stdbool.h>
+# include <glib.h>
+
+# ifdef __cplusplus
+#  define UTIL_C_BEG extern "C" {
+#  define UTIL_C_END };
+# else
+#  define UTIL_C_BEG
+#  define UTIL_C_END
+# endif
+
+UTIL_C_BEG
+
+/* ========================================================================= *
+ * Constants
+ * ========================================================================= */
+
+/* Expected: We get these from Makefile, or spec during rpm build */
+# ifndef  VERSION
+#  define VERSION    "0.0.0"
+# endif
+
+# ifndef  SYSCONFDIR
+#  define SYSCONFDIR "/etc"
+# endif
+
+# ifndef  LIBDIR
+#  define LIBDIR     "/usr/lib"
+# endif
+
+# ifndef  DATADIR
+#  define DATADIR    "/usr/share"
+# endif
+
+/* Config from: *.conf */
+# define CONFIG_DIRECTORY               SYSCONFDIR "/sailjail/config"
+# define CONFIG_EXTENSION               ".conf"
+# define CONFIG_PATTERN                 "[0-9][0-9]*" CONFIG_EXTENSION
+
+/* Users from: passwd */
+# define USERS_DIRECTORY                SYSCONFDIR
+# define USERS_EXTENSION                ""
+# define USERS_PATTERN                  "passwd" USERS_EXTENSION
+
+/* Permissions from: *.permission */
+# define PERMISSIONS_DIRECTORY          SYSCONFDIR "/sailjail/permissions"
+# define PERMISSIONS_EXTENSION          ".permission"
+# define PERMISSIONS_PATTERN            "[A-Z]*" PERMISSIONS_EXTENSION
+
+/* Applications  from: *.desktop */
+# define APPLICATIONS_DIRECTORY         DATADIR "/applications"
+# define APPLICATIONS_EXTENSION         ".desktop"
+# define APPLICATIONS_PATTERN           "*" APPLICATIONS_EXTENSION
+
+/* Settings from: *.settings */
+# define SETTINGS_DIRECTORY             LIBDIR "/sailjail/settings"
```

Comment:
This should be in /home/.system/

Reply:
It certainly shouldn't be in /usr/lib because this is changing data. I think FHS place for that would be in /var/lib. However root filesystem is not encrypted. It's better to keep this under /home partition which is encrypted and thus can't be read from turned off device. That's my thinking here.

Reply:
Moved under /home/.system and restricted directory accessibility.

@Tomin1 Haven't done After=home things before -> pay extra attention to .service changes ;-)

Reply:
Not sure if it is github acting up or what, but ... these threads behave in surprising ways. This was resolved elsewhere - I wonder what I've missed.

Reply:
Yeah, if you comment a line that was already commented before in a new review, then the comment ends up as a reply and also as a new comment in the new review. It's confusing.